src/web/api.rs

```diff
@@ -279,6 +285,112 @@ impl ApiService {
             session_id: self.serve_session.session_id(),
         })
     }
+
+    fn handle_api_create_assets(&self, request: Request<Body>) -> <Self as Service>::Future {
+        let session_id = self.serve_session.session_id();
+
+        let serve_tree = self.serve_session.tree_handle();
+        let vfs = self.serve_session.vfs_handle();
+
+        Box::new(request.into_body().concat2().and_then(move |body| {
```

Comment:
Damn. We've gotta refactor this code to start using async/await someday, lol.

Reply:
The errors I got when writing this were so confusing, please :cry: